### PR TITLE
qdma4: add pre-setup for slave bridge && enable st

### PR DESCRIPTION
**Pull request by karenx-xilinx**
- add streaming mode support
- add configuration setup for slave bridge

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/devices.h [hunk 1]
```diff
@@ -1482,6 +1482,12 @@ struct xocl_subdev_map {
 	 		.name = NODE_STM4,		\
 			.flags = IORESOURCE_MEM,	\
 		},					\
+		{					\
+			.start = 0x3000000,		\
+			.end = 0x3008000,		\
+	 		.name = NODE_QDMA4_CSR,		\
+			.flags = IORESOURCE_MEM,	\
+		},					\
```
Comment (houlz0507) [inline src/runtime_src/core/pcie/driver/linux/xocl/devices.h:1490]: Why do we need static define of XOCL_RES_QDMA4? Is QDMA4 only for 2RP platforms?

File: src/CMake/dkms.cmake [hunk 2]
```diff
@@ -184,6 +184,8 @@ SET (XRT_DKMS_DRIVER_SRCS
   xocl/subdev/xdma.c
   xocl/subdev/qdma.c
   xocl/subdev/qdma4.c
+  xocl/subdev/stmc.h
+  xocl/subdev/stmc.c
```
Comment (houlz0507) [inline src/CMake/dkms.cmake:188]: Looks like stmc is a new subdev. Why are not the subdev id and name defined in devices.h?

File: src/runtime_src/core/pcie/driver/linux/xocl/devices.h (hunk 1 above)
Comment (karenx-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/devices.h:1490]: Hmm, I am not aware of EQDMA for 1RP.

File: src/CMake/dkms.cmake (hunk 2 above)
Comment (karenx-xilinx) [reply src/CMake/dkms.cmake:188]: I have added NODE_STM4 in device.h in the previous check-ins.
Do i need to rename these 2 files to stm4.[ch]?

**Comment by karenx-xilinx** [comment]
withdraw, need to re-arrange some files.

**Comment by karenx-xilinx** [comment]
obselete

**Comment by karenx-xilinx** [comment]
obsolete

**Comment by karenx-xilinx** [comment]
obsolete

**Pull request closed without merging**